### PR TITLE
Moving some commands off of async stream

### PR DESCRIPTION
**Pull request by JosephTLyons**

### CODE REVIEW

File: crates/nu-cli/src/commands/mkdir.rs [hunk 1]
```diff
@@ -44,20 +44,20 @@ impl WholeStreamCommand for Mkdir {
...
     let registry = registry.clone();
-    let stream = async_stream! {
-        let name = args.call_info.name_tag.clone();
-        let shell_manager = args.shell_manager.clone();
-        let (args, _) = args.process(&registry).await?;
-        let mut result = shell_manager.mkdir(args, name)?;
+    let name = args.call_info.name_tag.clone();
+    let shell_manager = args.shell_manager.clone();
+    let (args, _) = args.process(&registry).await?;
+    let mut result = shell_manager.mkdir(args, name)?;
```
Comment (jonathandturner) [inline crates/nu-cli/src/commands/mkdir.rs:52]: I think you might be able to just call `shell_manager.mkdir(args, name)` and let the result of it be the return value of the function

File: crates/nu-cli/src/commands/rm.rs [hunk 2]
```diff
@@ -73,27 +73,28 @@ impl WholeStreamCommand for Remove {
...
+    let shell_manager = args.shell_manager.clone();
+    let (args, _): (RemoveArgs, _) = args.process(&registry).await?;
+    let mut result = if args.trash.item && args.permanent.item {
+        OutputStream::one(Err(ShellError::labeled_error(
+            "only one of --permanent and --trash can be used",
+            "conflicting flags",
+            name,
+        )))
+    } else {
+        shell_manager.rm(args, name)?
```
Comment (jonathandturner) [inline crates/nu-cli/src/commands/rm.rs:88]: Same here, if you doing use `?` and then for the case of the if above, returned `Ok(OutputStream...)`

**Comment by jonathandturner** [comment]
This looks good - I just had a couple small nits that might help you clean it up even more

File: crates/nu-cli/src/commands/mkdir.rs (hunk 1 above)
Comment (JosephTLyons) [reply crates/nu-cli/src/commands/mkdir.rs:52]: Oh nice, yeah I totally missed that.  Yeah what I was doing here was totally unnecessary.

**Comment by jonathandturner** [comment]
Thanks!

**Pull request merged**